aep/general/0121/aep.md

```diff
@@ -1,5 +1,155 @@
+has the collection of books that it publishes. A resource usually has fields,
+and resources may have any number of sub-resources (usually collections).
+
+**Note:** While there is some conceptual alignment between storage systems and
+APIs, a service with a resource-oriented API is not necessarily a database, and
+has enormous flexibility in how it interprets resources and methods. API
+designers **should not** expect that their API will be reflective of their
+database schema. In fact, having an API that is identical to the underlying
+database schema is actually an anti-pattern, as it tightly couples the surface
+to the underlying system.
+
+### Methods
+
+Resource-oriented APIs emphasize resources (data model) over the methods
+performed on those resources (functionality). A typical resource-oriented API
+exposes a large number of resources with a small number of methods on each
+resource. The methods can be either the standard methods ([Get][], [List][],
+[Create][], [Update][], [Delete][]), or [custom methods][].
+
+If the request to or the response from a standard method (or a custom method in
+the same _service_) **is** the resource or **contains** the resource, the
+resource schema for that resource across all methods **must** be the same.
+
+| Standard method | Request               | Response        |
+| --------------- | --------------------- | --------------- |
+| Create          | Contains the resource | Is the resource |
+| Get             | None                  | Is the resource |
+| Update          | Contains the resource | Is the resource |
+| Delete          | None                  | None            |
+| List            | None                  | Is the resource |
+
+_The table above describes each standard method's relationship to the resource,
+where "None" indicates that the resource neither **is** nor **is contained** in
+the request or the response_
+
+A resource **must** support at minimum [Get][]: clients must be able to
+validate the state of resources after performing a mutation such as [Create][],
+[Update][], or [Delete][].
+
+A resource **must** also support [List][], except for [singleton resources][]
+where more than one resource is not possible.
+
+**Note:** A custom method in resource-oriented design does _not_ entail
+defining a new or custom HTTP verb. Custom methods use traditional HTTP verbs
+(usually `POST`) and define the custom verb in the URI.
+
+APIs **should** prefer standard methods over custom methods; the purpose of
+custom methods is to define functionality that does not cleanly map to any of
+the standard methods. Custom methods offer the same design freedom as
+traditional RPC APIs, which can be used to implement common programming
+patterns, such as database transactions, import and export, or data analysis.
+
+### Strong Consistency
+
+For methods that operate on the [management plane][], the completion of those
+operations (either successful or with an error, LRO or synchronous) **must**
```

Comment:
Might need to define/explain "LRO".

Reply:
ack, added a link. Does that work?

Reply:
I added a related comment [here](https://github.com/aep-dev/aep.dev/pull/84#discussion_r1518105083) (couldn't add a suggestion to this thread because the comment is not on the latest commit).

Reply:
agreed! thanks for adding the suggestions.